.github/workflows/tests.yml

```diff
@@ -1,6 +1,21 @@
 name: tests
 
-on: [push, pull_request, workflow_dispatch]
+on:
+  push:
+    paths:
+      - "**.nim"
```

Comment:
See https://docs.github.com/en/actions/reference/workflow-syntax-for-github-actions#patterns-to-match-file-paths for examples of what this pattern does